src/service.rs

```diff
@@ -108,59 +108,54 @@ impl Account {
     pub fn signup(client: Client, user: &User, password: &str) -> Result<Self> {
         super::init()?;
 
-        let salt = randombytes(32);
+        let salt = randombytes_array();
         let main_key = derive_key(&salt, password)?;
 
-        Self::signup_common(client, user, main_key, salt)
+        Self::signup_common(client, user, main_key, &salt)
     }
 
     /// Creates a new user on the server and returns a handle to it. The user is authenticated
     /// using a cryptographically secure random 32-byte `main_key` instead of a password.
     pub fn signup_key(client: Client, user: &User, main_key: &[u8]) -> Result<Self> {
         super::init()?;
 
-        if main_key.len() != SYMMETRIC_KEY_SIZE {
-            return Err(Error::ProgrammingError(
-                "Key should be exactly 32 bytes long.",
-            ));
-        }
-
-        let salt = randombytes(32);
-        let main_key = main_key.to_vec();
+        // TODO: change argument type to [u8; 32]
+        let main_key: [u8; 32] = main_key
+            .try_into()
+            .map_err(|_| Error::ProgrammingError("Key should be exactly 32 bytes long."))?;
 
-        Self::signup_common(client, user, main_key, salt)
+        Self::signup_common(client, user, main_key, &[])
```

Comment:
I'm not entirely sure about this one. When signing up with a key, the salt is unused, so the straightforward solution would be to never generate one - but doing it leaks the information whether an account uses a hashed password or a key. Is that a problem?

Reply:
It's probably not a major problem, but it's unnecessary. Why not just generate a salt and send it?

Reply:
Done.